de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/SarosSessionManager.java

```diff
@@ -114,6 +115,16 @@
 
     private final ProjectNegotiationObservable currentProjectNegotiations;
 
+    /** to queue resources for the next project negotiation **/
+    private final Map<IProject, List<IResource>> nextSharedResources = new HashMap<IProject, List<IResource>>();
+    /** to set the thread for handling the next project negotiation **/
+    private volatile boolean nextSharedResourcesWorker = false;
+    /**
+     * to notify a thread waiting for completion of all running project
```

Comment:
See my comment before this review, move the code to a dedicated class. The comments aint that helpful, even if you look a the code where all these objects are used.

Reply:
I would prefer to keep it at this place for now and change it with the refactoring of the package.